### ISSUE TITLE
New Mode 2: Helper Drives on while filling

### ISSUE THREAD

**Issue by enderwiggin07** [Courseplay/courseplay#5492]
Not posting logs because I think this is an obvious one, while unloading a stopped combine, the helper obeys the "Drive on at:" setting even if the combine is still unloading.  I believe in the past the "Drive on" setting was only used while parked.  While loading the helper should accept up to his full capacity before driving on.

This new mode 2 is beautiful to watch I have to say, mode 2 is really the heart and soul of courseplay.  I appreciate what you've made.

**Comment by Goof245** [comment]
In the past I think he used drive on while filling, but only if there was no grain going into the tank. So you'd set the % such that it's the most the tank can hold while still leaving room for a full combine's worth of grain.

Once he'd go over the drive on %, he'd wait until the combine had finished unloading and then go to the unload course.

**Comment by enderwiggin07** [comment]
Yeah that sounds exactly as I recall.  At any rate right now it drives at
(example) 90% even if the combine is still unloading.  But only for a
stopped combine (after doing the maneuver to get the pipe out of fruit) but
with a moving combine (pipe out of fruit) it seems to work correctly
already

**Comment by Tensuko** [comment]
So check 'drive on %' only if Combine is empty or Trailer is full ?

**Comment by polosubtil** [comment]
Hi @Tensuko 
I noticed that too but I checked with the latest versions of 6.02
I think you are correct in your last statement, the "drive on %" should be checked only after the combine is empty

**Comment by enderwiggin07** [comment]
> 
> 
> So check 'drive on %' only if Combine is empty or Trailer is full ?

Exactly.  It looks like it's doing this as long as the combine is moving already, that is working correctly.

But when the combine has stopped and moved the pipe out of fruit, should be changed to what you said

**Comment by Tensuko** [comment]
while unloading, and it passes the set % while unloading and combine is not empty, keep loading... not sure if we rly will do it like that as you could just increase the % to 99 or 100... have to discuss this.

**Comment by polosubtil** [comment]
Hi @Tensuko 
It's not the same thing. If you are unloading into the trailer, you do it until the trailer is full or the combine is empty. This setting is there for a better organization at the end of the unloading, that's all.
The point of this setting is to ask the trailer to run the course so it doesn't wait for the combine to call when the trailer is almost full. For example I always set that setting to 90% (28000 liters trailers), if you are above that value, you have less than 2800 liters left which isn't enough for most combines, which means you lose a lot of time as you trailer will start the course while the combine is almost full
This is how it use to work with the previous version 6.01 this is why we are asking, but this is not a proper issue as long as mode 2 is running ;)

**Comment by Goof245** [comment]
Setting it that close to full defeats the purpose of setting it at all. The main reason this setting is useful is you can prevent the unloader turning up to unload the combine with an auger wagon that's 99/98% full...

**Comment by Tensuko** [comment]
@polosubtil I get the problem but we have to talk about how we want to manage this anyway so just wait for it...
